apps/fz_http/lib/fz_http/allow_rules.ex

```diff
@@ -4,34 +4,96 @@ defmodule FzHttp.AllowRules do
   """
 
   import Ecto.Query, warn: false
-  alias FzHttp.{AllowRules.AllowRule, Gateways.Gateway, Repo, Users.User}
+  alias FzHttp.{AllowRules.AllowRule, Gateways.Gateway, Repo, Users.User, Events, Telemetry}
 
   def create_allow_rule(attrs \\ %{}) do
+    case %AllowRule{}
+         |> AllowRule.changeset(attrs)
+         |> Repo.insert() do
+      {:ok, rule} ->
+        Events.add(rule)
+        Telemetry.add_rule()
+        {:ok, rule}
+
+      {:error, changeset} ->
+        dbg(changeset)
+        {:error, changeset}
+    end
+  end
+
+  def as_setting(rule) do
+    user_uuid =
+      case Repo.preload(rule, :user).user do
+        nil -> nil
+        user -> user.uuid
+      end
+
+    %{
+      dst: rule.destination,
+      user_uuid: user_uuid,
+      port_range: add_port_settings(rule)
+    }
+  end
+
+  defp add_port_settings(%{
+         port_range_start: range_start,
+         port_range_end: range_end,
+         protocol: protocol
+       })
+       when range_start != nil do
+    %{
+      range_start: range_start,
+      range_end: range_end,
+      protocol: protocol
+    }
+  end
+
+  defp add_port_settings(_), do: nil
+
+  def delete_allow_rule(%AllowRule{} = allow_rule) do
+    case Repo.delete(allow_rule) do
+      {:ok, rule} ->
+        Events.delete(rule)
+        Telemetry.delete_rule()
+
```

Comment:
There's somewhat of a convention for these context modules to return `{:ok, record}` tuples for successful operations and `{:error, changeset}` for errors.

```suggestion
        {:ok, rule}
```